src/symbolic_dimensions.jl

```diff
@@ -1,21 +1,20 @@
 import .Units: UNIT_SYMBOLS, UNIT_MAPPING, UNIT_VALUES
 import .Constants: CONSTANT_SYMBOLS, CONSTANT_MAPPING, CONSTANT_VALUES
 
-
 const SYMBOL_CONFLICTS = intersect(UNIT_SYMBOLS, CONSTANT_SYMBOLS)
 
 disambiguate_symbol(s) = s in SYMBOL_CONFLICTS ? Symbol(s, :_constant) : s
 
-const INDEX_TYPE = UInt8
 # Prefer units over constants:
 # For example, this means we can't have a symbolic Planck's constant,
 # as it is just "hours" (h), which is more common.
-const ALL_SYMBOLS = (
-    UNIT_SYMBOLS...,
-    disambiguate_symbol.(CONSTANT_SYMBOLS)...
-)
-const ALL_VALUES = (UNIT_VALUES..., CONSTANT_VALUES...)
-const ALL_MAPPING = NamedTuple{ALL_SYMBOLS}(INDEX_TYPE(1):INDEX_TYPE(length(ALL_SYMBOLS)))
+const INDEX_TYPE = UInt16
+# Prefer units over constants:
+# For example, this means we can't have a symbolic Planck's constant,
+# as it is just "hours" (h), which is more common.
+const ALL_SYMBOLS = [UNIT_SYMBOLS..., disambiguate_symbol.(CONSTANT_SYMBOLS)...]
+const ALL_VALUES = [UNIT_VALUES..., CONSTANT_VALUES...]
+const ALL_MAPPING = Dict(ALL_SYMBOLS .=> (INDEX_TYPE(1):INDEX_TYPE(length(ALL_SYMBOLS))))
```

Comment:
The broadcasting is not needed:
```suggestion
const ALL_MAPPING = Dict(s => INDEX_TYPE(i) for (i, s) in enumerate(ALL_SYMBOLS))
```

Reply:
When I benchmarked, broadcasting was much faster than enumerating. So I switched.

This is what I ran:
```julia
julia> k = [Symbol(randstring(10)) for i in 1:100]
julia>INDEX_TYPE = UInt16
julia> @benchmark ALL_MAPPING = Dict($k .=> (INDEX_TYPE(1):INDEX_TYPE(length($k))))
BenchmarkTools.Trial: 10000 samples with 9 evaluations.
 Range (min … max):  1.878 μs … 411.511 μs  ┊ GC (min … max): 0.00% … 98.51%
 Time  (median):     2.100 μs               ┊ GC (median):    0.00%
 Time  (mean ± σ):   3.013 μs ±   9.624 μs  ┊ GC (mean ± σ):  9.70% ±  3.10%

  ▇█▄▂▃▃▄▃▂▂▃▁                              ▂▂                ▂
  ██████████████▇▇▅▆▅▆▄▄▄▄▄▄▃▃▄▃▁▁▃▁▁▁▃▁▄▆▇████▇▅▇▆▆▆▄▅▅▅▄▅▃▅ █
  1.88 μs      Histogram: log(frequency) by time      11.8 μs <

 Memory estimate: 5.08 KiB, allocs estimate: 10.

julia> @benchmark  ALL_MAPPING = Dict(s => INDEX_TYPE(i) for (i, s) in enumerate($k))
BenchmarkTools.Trial: 10000 samples with 1 evaluation.
 Range (min … max):  30.000 μs …  3.332 ms  ┊ GC (min … max): 0.00% … 97.70%
 Time  (median):     31.000 μs              ┊ GC (median):    0.00%
 Time  (mean ± σ):   33.411 μs ± 33.523 μs  ┊ GC (mean ± σ):  0.97% ±  0.98%

  ▃█▇▇▅▄▃▃▂▂▂                                     ▁▂▁▁▁       ▂
  ██████████████▇▇▆▆▆▅▆▆▅▅▅▄▆▆▆▆▅▅▆▆▆▇▆▇█▇▇█▇▇▇██▇█████▇██▆▆▆ █
  30 μs        Histogram: log(frequency) by time      51.5 μs <

 Memory estimate: 7.95 KiB, allocs estimate: 116.
```

Reply:
The problem is that you benchmark directly with the global `INDEX_TYPE`. This leads to incorrect results (an indication are the allocations in the second case). For instance, I get
```julia
julia> using BenchmarkTools

julia> k = [Symbol(randstring(10)) for i in 1:100];

julia> INDEX_TYPE = UInt16
UInt16

julia> function f(k, ::Type{T}) where {T}
           Dict(k .=> (T(1):T(length(k))))
       end
f (generic function with 1 method)

julia> function g(k, ::Type{T}) where {T}
           Dict(s => T(i) for (i, s) in enumerate(k))
       end
g (generic function with 1 method)

julia> function h(k, ::Type{T}) where {T}
           Dict(map((i, s) -> s => T(i), 1:length(k), k))
       end
h (generic function with 1 method)

julia> f(k, INDEX_TYPE) == g(k, INDEX_TYPE) == h(k, INDEX_TYPE)
true

julia> @benchmark f($k, $INDEX_TYPE)
BenchmarkTools.Trial: 10000 samples with 14 evaluations.
 Range (min … max):  961.286 ns … 99.491 μs  ┊ GC (min … max): 0.00% … 97.89%
 Time  (median):       1.116 μs              ┊ GC (median):    0.00%
 Time  (mean ± σ):     1.243 μs ±  3.204 μs  ┊ GC (mean ± σ):  9.84% ±  3.78%

      ▆▄▇▂     ▂▂▃█▄█▂▄
  ▁▂▆██████▇▅▇▆████████▇▅▆▄▄▃▂▂▂▂▂▂▂▂▂▂▁▁▂▂▂▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁ ▃
  961 ns          Histogram: frequency by time         1.57 μs <

 Memory estimate: 5.03 KiB, allocs estimate: 8.

julia> @benchmark g($k, $INDEX_TYPE)
BenchmarkTools.Trial: 10000 samples with 69 evaluations.
 Range (min … max):  837.565 ns … 112.315 μs  ┊ GC (min … max):  0.00% … 98.62%
 Time  (median):       1.010 μs               ┊ GC (median):     0.00%
 Time  (mean ± σ):     1.194 μs ±   2.656 μs  ┊ GC (mean ± σ):  14.97% ±  6.82%

    ▂▁             ▄▃██▆▆▄▆▃▂▁
  ▂███▅▂▂▂▂▃▃▂▃▃▃▆████████████▇█▇▇▆▅▅▄▅▄▃▃▃▃▂▂▃▃▂▂▂▂▂▂▂▂▂▁▁▁▁▁▁ ▄
  838 ns           Histogram: frequency by time          1.3 μs <

 Memory estimate: 3.27 KiB, allocs estimate: 7.

julia> @benchmark h($k, $INDEX_TYPE)
BenchmarkTools.Trial: 10000 samples with 10 evaluations.
 Range (min … max):  991.700 ns … 169.917 μs  ┊ GC (min … max):  0.00% … 98.46%
 Time  (median):       1.171 μs               ┊ GC (median):     0.00%
 Time  (mean ± σ):     1.340 μs ±   4.559 μs  ┊ GC (mean ± σ):  10.60% ±  3.11%

     ▂    ▃▃▆█▅█▃▄▁
  ▂▄▇███▇▇█████████▆▆▄▃▃▃▃▃▃▂▃▃▂▃▂▃▃▃▃▃▃▃▂▂▂▂▂▂▂▂▂▁▂▂▂▂▂▂▂▂▂▂▂▂ ▄
  992 ns           Histogram: frequency by time          1.9 μs <

 Memory estimate: 5.03 KiB, allocs estimate: 8.
```

Reply:
Resovled https://github.com/SymbolicML/DynamicQuantities.jl/pull/107/commits/bfc699ffbe5f0e0d882f168de2b8bd92fec9941f